### ISSUE TITLE
Please add a serverless option (Oasis requires customer to guess + manage capacity)

### ISSUE THREAD

**Issue by bionicles** [arangodb/arangodb#10913]
__Affected feature__:
Installation + Onboarding

__Size of your Dataset on disk__:
Unknown and variable startup prototype(s)

__Replication Factor & Number of Shards (Cluster only)__:
Unknown but HA is critical for a startups

## Steps to reproduce

1. Try to use arangodb
2. You need to manage servers
3. Try ArangoDB Oasis
4. You need to specify a fixed capacity to run in advance

__Problem__:
It's hard to get started with ArangoDB for startups because there's so much devops involved and the current solution requires you to guess your capacity needs in advance, run that capacity 24/7 and manually monitor the usage and manually jump through a bunch of hoops to scale the capacity up and down. 

We would not use Oasis now because we can just use Terraform or AWS Marketplace ... it says "fully managed" but the hardest part (scaling) is still left to the customer to manage! 

We would definitely use ArangoDB Oasis if there were a **serverless option with on-demand pricing** like [Amazon DynamoDB On Demand Capacity](https://aws.amazon.com/dynamodb/pricing/on-demand/) or [Microsoft Azure CosmosDB Autopilot](https://azure.microsoft.com/en-us/pricing/details/cosmos-db/) because then we could waste less energy and money for little prototypes while also scaling automatically. 

__Expected result__:
Go to ArangoDB site
Sign up + provide credit card number
Make a graph + get an endpoint & credentials
Pay for what we use

TLDR No one has a great serverless graph database and many customers would use and benefit from an ArangoDB Serverless pricing+scaling option

**Comment by MrPieces** [comment]
Hi Bion,

thanks for your detailed feedback and I fully understand, that especially for a startup like yours it is important to think about the time when your idea is going through the roof. We think about the same thing for our users. 

Before the Oasis project started, we spoke with many people we know, also people who are using offerings like Dynamo or Microsoft on demand and it turned out that these services appear very cheap at the beginning but can become comparably very expensive once you scale. In this case, a server based pricing can be much cheaper. I'm not saying that our current offering is the pinnacle of all wisdom :)

I'm not sure I agree with your statement, that the hardest part is scaling, seen from a database perspective. This highly depends on your use case, queries, data model and many other factors. A very time consuming part and also quite challenging is to keep a self hosted system available and fix any problem that might occur in production. In my point of view, this is the big strength of a managed cloud service, where someone else (ArangoDB experts) are making sure your database keeps running and if not fix the problem... so you can focus on scaling your business. Getting started with this kind of insurance for $150/month or $0.21/hour is pretty cheap in my eyes.  In addition, you can always scale up and down your deployment in terms of nodes, RAM, storage and other parameters. Yes this is currently a user induced decision and we may add an automatic functionality in the future.

Of course, I do not know anything about your use case, queries, data model or other aspects important to keep in mind when scaling an application up. A clear benefit of Oasis is, that it also supports the special features of the enterprise edition of ArangoDB for scaling with the supported data models like SmartGraphs or SmartJoins (yeah, I now not really smart names ) at no additional costs. So if you are already thinking about a scale that is going through the roof, then I recommend to keep in mind that running queries like traversals or joins is not easily done against data residing on different machines but with SmartGraphs or SmartJoins pretty efficient.

**Comment by bionicles** [comment]
@MrPieces @OmarAyo my concern is mainly one of not having to choose a capacity level, it's a guessing game which could be driven directly by data, automatically.

$150 a month seems cheap but if there is no autoscaling then someone at the startup needs to monitor the database capacity 24/7 and manually go into some console

is it difficult for arangodb oasis to script this? if resource usage > 95% for 5 minutes, scale up, if resource usage < 5% for some time, scale down. Easy. The system supports kubernetes and multitenancy already kubernetes has pod autroscaling. 

Simply put, when I see these sign up pages for "managed service" then I'm taken to a screen where I choose some number of gigabytes, cpus, it just isn't fun. I have to guess how many users I have, how many documents per user? how many bytes per document? That's a different number for each of the 1234567890 NoSQL databases. To be blunt, a "managed service" which does not manage scaling usually makes me chuckle to myself and move on, because it doesn't manage the scaling, so I still have to watch that stuff. That's exactly the administration and guesswork I want the service to manage! How can the possibly model the resource needs for a complex system like a database better than the company who makes it? 

I'm also passionate about the need for openCypheer MATCH syntax because you can write 1 liners in RedisGraph or Neo4j which take 3 - 5 lines of AQL
https://github.com/arangodb/arangodb/issues/10965 

And the docs for ArangoDB drivers require at least 6 clicks to get from arangodb.com to the page about Graph Vertices. But the graph vertices docs don't cover vertex crud, they cover vertexCollection CRUD. You click on Vertices and you get VertexCollection operations, and you click on VertexCollection and you get Vertex operations. The entire set of graph documentation is scattered across about 10 short pages so I can't CTRL-F... and since I need to automate CI/CD, I need to write a script to create a graph, which means I need edge definitions, which means I need an edge collection, which means I need a vertex collection. (mentioned here https://github.com/arangodb/docs/issues/312 and was told basically "we decided to keep the docs scattered across many pages with poor names and won't fix the docs being nested 7 levels deep on the site" 

So to sign up and use arangodb I need to:

1. Guess my capacity
2. Open 10+ tabs of documentation and remember what's on each tab since I can't CTRL-F
3. Give up on the openCypher MATCH syntax
4. Write a script to Create A Graph
5. Oh, I can't, I need an edge collection, how do you make that? Click Edge Collection. No wait, that's how you create edges. 
6. Click Edges. Now you see how to create an edge collection. Oh, you need a vertex collection, do you click where it says "vertex collection" or "vertices" ? 
7. Test the script ... 

IDK, I guess I should just make one graph, manually, and not version control my schema, but as a devops engineer + entrepreneur that's super risky. If I can't run a script to reproduce the database, what happens if something goes wrong, and what do I sell when I sell my startup? A customer wants help setting up their own graph. Now I have to manually recreate a schema in a NoSQL database, and manually scale my capacity, which entails manual monitoring of capacity usage? 

At that point, I'd rather just navigate to Fauna.com or aws.amazon.com and sign up.

**Comment by homerjam** [comment]
> TLDR No one has a great serverless graph database

https://docs.microsoft.com/en-us/azure/cosmos-db/serverless

I'm adding this as a reference, I'd love to use Arangodb in a hobby/side project capacity so I can learn it and confidently recommend it for larger, paying projects. However the overheads of setup/management for a single node or the cost for Oasis mean I'm yet to really experiment with it.

**Comment by doorkey** [comment]
Hey guys. I second the Binicles comments. I have trusted ArangoDB for my startup as well and now I am kinda stuck while there are many incredibly cheap options are out there which are serverless. MongoDB just launched the serverless product which is really making me to switch from ArangoDB. There is new player Fauna which offers much cheaper serverless multi-model db options. 

I fell in love with Arango coz of its innovation and agility. I think you should consider offering Serverless option otherwise you will lose a lot of your business and even users. I am already planning a move to change dbs which is gonna be another cost but in long-run is still cheaper.

**Comment by ewoutp** [comment]
@homerjam @doorkey We fully understand the desire for flexible usage of all infrastructure, including the database.
Earlier this year, Oasis introduced Developer deployments. These are single instances with a much lower price point than the usual cluster deployments.

We are also actively looking into several auto-sizing features. The first will be around disk size, but automatic adjustments of memory and CPU resources will follow soon.
